src/Raven.Server/Documents/ETL/Providers/AI/Extensions/DimensionalityHandler.cs

```diff
@@ -0,0 +1,84 @@
+﻿using System.IO;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Raven.Client.Documents.Conventions;
+using Raven.Client.Json;
+using Sparrow.Json;
+using Sparrow.Json.Parsing;
+using Sparrow.Server.Json.Sync;
+
+namespace Raven.Server.Documents.ETL.Providers.AI.Extensions;
+
+public class DimensionalityHandler : DelegatingHandler
+{
+    private const string BatchEmbedContents = ":batchEmbedContents";
+    private const string EmbedContent = ":embedContent";
+    private const string OutputDimensionality = "output_dimensionality";
+    private const string Requests = "requests";
+    private const string MediaType = "application/json";
+    private const string BlittableDocumentId = "requestBody/json";
+
+    private static readonly DocumentConventions ConventionsToUse = new() { UseHttpCompression = false };
```

Comment:
Why is that disabled?

Reply:
Thx, it wasn't working, but now I realized that we use zstd as default. Switched to gzip.